File: fixed_array.rb
class FixedArray
  def initialize(size)
    @array = Array.new(size)
  end

  def [](index)
    check_bounds(index)
    @array[index]
  end

  def []=(index, value)
    check_bounds(index)
    @array[index] = value
  end

  def size
    @array.size
  end

  private

  def check_bounds(index)
    raise IndexError, "Invalid index: #{index}" if index < 0 || index >= size
  end
end

File: javascript/lib/fixed-array.js
"use strict";

var IndexError = require("./index-error");

function FixedArray(size) {
    this._array = new Array(size);
    this._array.fill(null);
}

FixedArray.prototype.get = function(index) {
    this._checkBounds(index);
    return this._array[index];
};

FixedArray.prototype.set = function(index, value) {
    this._checkBounds(index);
    this._array[index] = value;
};

FixedArray.prototype.size = function() {
    return this._array.length;
};

FixedArray.prototype._checkBounds = function(index) {
    if (index < 0 || index >= this.size()) {
        throw new IndexError("Invalid index: " + index);
    }
};

module.exports = FixedArray;
